apps/sms/js/contacts.js

```diff
@@ -98,29 +98,26 @@
     findBy: function contacts_findBy(filter, callback) {
       var lower = [];
       var filterValue = (filter.filterValue || '').trim();
-      var terms, request;
+      var terms;
 
       if (!navigator.mozContacts || !filterValue.length) {
-        setTimeout(function() {
-          callback(
-            /**
-             * Bailout Strategy
-             *
-             * 1. A _missing_ filter.filterValue will results in an
-             *    error when passed to mozContacts.find, bailing out
-             *    early avoids the additional overhead of making a
-             *    request that we know will fail.
-             *
-             * 2. Missing navigator.mozContacts API or empty but
-             *    present filter.filterValue make the app useless,
-             *    so prevent unpredictable state by making the
-             *    result a no-op.
-             *
-             */
-            typeof filter.filterValue === 'undefined' ? null : [], {}
-          );
-        });
-        return;
+        /**
+         * Bailout Strategy
+         *
+         * 1. A _missing_ filter.filterValue will results in an
+         *    error when passed to mozContacts.find, bailing out
+         *    early avoids the additional overhead of making a
+         *    request that we know will fail.
+         *
+         * 2. Missing navigator.mozContacts API or empty but
+         *    present filter.filterValue make the app useless,
+         *    so prevent unpredictable state by making the
+         *    result a no-op.
+         *
+         */
+        return Promise.resolve(
```

Comment:
Do we really need this ```null``` return value? I'm not sure if we have code somewhere that has drastically different behavior for empty array and null result.

IMO it just makes code that consumes API more complex - we always should check ```contacts && contacts.length``` or ```contacts || []``` like in methods below. Usually list-based API methods always return an array. In our case if we have really unrecoverable error we should just reject promise.

What do you think?